common/models/checkpoint.js

```diff
@@ -27,43 +27,46 @@ module.exports = function(Checkpoint) {
    * Get the current checkpoint id
    * @callback {Function} callback
    * @param {Error} err
-   * @param {Number} checkpointId The current checkpoint id
+   * @param {Number} checkpoint The current checkpoint seq
    */
-
   Checkpoint.current = function(cb) {
     var Checkpoint = this;
-    this.find({
-      limit: 1,
-      order: 'seq DESC'
-    }, function(err, checkpoints) {
-      if (err) return cb(err);
-      var checkpoint = checkpoints[0];
-      if (checkpoint) {
-        cb(null, checkpoint.seq);
-      } else {
-        Checkpoint.create({ seq: 1 }, function(err, checkpoint) {
-          if (err) return cb(err);
-          cb(null, checkpoint.seq);
-        });
-      }
+    Checkpoint._getSingleton(function(err, cp) {
+      cb(err, cp.seq);
     });
   };
 
-  Checkpoint.observe('before save', function(ctx, next) {
-    if (!ctx.instance) {
-      // Example: Checkpoint.updateAll() and Checkpoint.updateOrCreate()
-      return next(new Error('Checkpoint does not support partial updates.'));
-    }
+  Checkpoint._getSingleton = function(cb) {
+    var query = {limit: 1}; // match all instances, return only one
+    var initialData = {seq: 1};
+    this.findOrCreate(query, initialData, cb);
+  };
 
-    var model = ctx.instance;
-    if (!model.getId() && model.seq === undefined) {
-      model.constructor.current(function(err, seq) {
-        if (err) return next(err);
-        model.seq = seq + 1;
-        next();
+  /**
+   * Increase the current checkpoint If it already exists otherwise initialize it
+   * @callback {Function} callback
+   * @param {Error} err
+   * @param {Object} checkpoint The current checkpoint
+   */
+  Checkpoint.bump = function(cb) {
```

Comment:
I feel `Checkpoint.bump` may not be descriptive enough. How about `Checkpoint.bumpLastSeq` or `Checkpoint.createNextSeq`?

/cc @ritch could you suggest a better name?